modules/gke/gsas.tf

```diff
@@ -9,7 +9,9 @@ resource "google_project_iam_member" "gke_nodes" {
   for_each = toset([
     "roles/logging.logWriter",
     "roles/monitoring.metricWriter",
-    "roles/monitoring.viewer"
+    "roles/monitoring.viewer",
+    "roles/stackdriver.resourceMetadata.writer",
+    "roles/artifactregistry.reader",
```

Comment:
Instead of adding this here, could we use the least privileges principle by doing this instead?
```
resource "google_artifact_registry_repository_iam_member" "gar_containers_reader" {
  location    = local.registry_location
  repository  = local.registry_id
  role        = "roles/artifactregistry.reader"
  member      = "serviceAccount:${google_service_account.gke_nodes.email}"
  
  depends_on  = [google_artifact_registry_repository.repo]
}
```